angular/src/app/services/wallet-manager.ts

```diff
@@ -797,7 +800,22 @@ export class WalletManager {
 
     // If the wallet type is restored, force an index process to restore the state.
     if (wallet.restored && runIndexIfRestored == true) {
-      this.communication.send(this.communication.createMessage('index', { force: true }, 'background'));
+      this.communication.send(this.communication.createMessage('index', {force: true}, 'background'));
+    }
+
+    if (network.smartContractSupport) {
+      await this.LoadStandardTokensForAccountAsync(network, this.getAddressByIndex(account, 0, 0));
+    }
+  }
+
+  async LoadStandardTokensForAccountAsync(network: Network, address: string) {
+    const indexerUrl = this.networkLoader.getServer(network.id, this.settings.values.server, this.settings.values.indexer);
+
+    const tokens = await axios.get(`${indexerUrl}/api/query/${network.name}/tokens/${address}`);
+    if (tokens.data.items) {
+      for (let token of tokens.data.items)
+        this.tokensStore.set(token.name, token);
```

Comment:
Does this distinguish between wallets and accounts? Are all tokens stored globally and accessible for all wallets? Should be filtered based upon accounts.

Reply:
Changed the tokens to be stored with the account identifier as key holding the list of tokens under it